pandas/types/cast.py

```diff
@@ -638,9 +651,28 @@ def _soft_convert_objects(values, datetime=True, numeric=True, timedelta=True,
     if datetime:
         values = lib.maybe_convert_objects(values, convert_datetime=datetime)
 
-    if timedelta and is_object_dtype(values.dtype):
-        # Object check to ensure only run if previous did not convert
-        values = lib.maybe_convert_objects(values, convert_timedelta=timedelta)
+    if is_object_dtype(values.dtype):
```

Comment:
I don't think we need to add ``period`` conversions here. It is quite difficult to know whether dates or periods are requested (yes, obviously the user is providing the flag), but in reality they should just use ``.to_datetime`` (or ``.to_period``)? maybe better just create that.